test/test_katello/test_deb_tracer.py

```diff
@@ -0,0 +1,176 @@
+NEEDRESTART-SVC: systemd-logind.service
+NEEDRESTART-SVC: systemd-timesyncd.service
+NEEDRESTART-SVC: unattended-upgrades.service
+NEEDRESTART-SVC: user@0.service """
+
+        output_list = needrestart_output.split("\n")
+        m.return_value = needrestart_output
+        self.assertEqual(needrestart(), output_list)
+        self.assertEqual(m.call_args.args, (["needrestart", "-b"],))
+        self.assertEqual(m.call_args.kwargs, {"universal_newlines": True})
+        self.assertEqual(len(collect_apps()), 12)
+
+    @patch.object(subprocess, "check_output")
+    def test_needrestart_exception_file_not_found(self, m):
+        m.side_effect = OSError()
+        with self.assertRaises(SystemExit) as cm:
+            needrestart()
+        the_exception = cm.exception
+        self.assertEqual(the_exception.code, 1)
+
+
+class TestCollectApps(unittest.TestCase):
+    @unittest.skipIf(sys.version_info[0] == 3, "not compatible with Python 3")
+    @patch("katello.deb_tracer.spawn.find_executable")
+    def test_has_needrestart_not_python2(self, m):
+        m.return_value = None
+        self.assertEqual(has_needrestart(), False)
+        self.assertEqual(m.call_args.args, ("needrestart",))
+
+    @unittest.skipIf(sys.version_info[0] < 3, "not compatible with Python 2")
+    @patch("katello.deb_tracer.which")
+    def test_has_needrestart_not_python3(self, m):
+        m.return_value = None
+        self.assertEqual(has_needrestart(), False)
+        self.assertEqual(m.call_args.args, ("needrestart",))
+
+    @unittest.skipIf(sys.version_info[0] == 3, "not compatible with Python 3")
+    @patch("katello.deb_tracer.spawn.find_executable")
+    def test_has_needrestart_ok_python2(self, m):
+        m.return_value = "/usr/sbin/needrestart"
+        self.assertEqual(has_needrestart(), True)
+        self.assertEqual(m.call_args.args, ("needrestart",))
+
+    @unittest.skipIf(sys.version_info[0] < 3, "not compatible with Python 2")
+    @patch("katello.deb_tracer.which")
+    def test_has_needrestart_ok_python3(self, m):
+        m.return_value = "/usr/sbin/needrestart"
+        self.assertEqual(has_needrestart(), True)
+        self.assertEqual(m.call_args.args, ("needrestart",))
+
+
+class TestUseFlag(unittest.TestCase):
+    @patch("katello.deb_tracer.path.isfile")
+    def test_reboot_needed_flag(self, m):
+        m.return_value = True
+        apps = use_flag()
+        self.assertEqual(type(apps), list)
+        self.assertTrue(len(apps) == 1)
+        self.assertIsInstance(apps[0], AptTracerApp)
+        # Validate kernel
+        self.assertEqual(apps[0].name, "kernel")
+        self.assertEqual(apps[0].helper, "You will have to reboot your computer")
+        self.assertEqual(apps[0].type, "static")
+
+    @patch("katello.deb_tracer.path.isfile")
+    def test_not_reboot_needed_flag(self, m):
+        m.return_value = False
+        apps = use_flag()
+        self.assertEqual(type(apps), list)
+        self.assertTrue(len(apps) == 0)
```

Comment:
`assertTrue` should be avoided for anything but booleans. That's because you don't really see what the actual value is when it's false. So I'd suggest to just assert that it's equal to an empty list instead:
```suggestion
        self.assertEqual(apps, [])
```